File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/AudienceStateEventId.java
package org.dddml.wms.domain;


public class AudienceStateEventId
{
    private String clientId;

    public String getClientId()
    {
        return this.clientId;
    }

    public void setClientId(String clientId)
    {
        this.clientId = clientId;
    }

    private Long version;

    public Long getVersion()
    {
        return this.version;
    }

    public void setVersion(Long version)
    {
        this.version = version;
    }

    AudienceStateEventId ()
    {
    }

    public AudienceStateEventId (String clientId, Long version)
    {
        this.clientId = clientId;
        this.version = version;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AudienceStateEventId other = (AudienceStateEventId)obj;
        return true 
            && (clientId == other.clientId || (clientId != null && clientId.equals(other.clientId)))
            && (version == other.version || (version != null && version.equals(other.version)))
            ;
    }

    @Override
    public int hashCode()
    {
        int hash = 0;
        if (this.clientId != null) {
            hash += 13 * this.clientId.hashCode();
        }
        if (this.version != null) {
            hash += 13 * this.version.hashCode();
        }
        return hash;
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AudienceStateEventId.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\StringIdInterface;

class AudienceStateEventId implements StringIdInterface
{
    /**
     * @Type("string")
     */
    private $clientId;

    /**
     * @return string
     */
    public function getClientId()
    {
        return $this->clientId;
    }

    /**
     * @param string $clientId
     */
    public function setClientId($clientId)
    {
        $this->clientId = $clientId;
    }

    /**
     * @Type("Dddml\Serializer\Type\Long")
     */
    private $version;

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->version;
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->version = $version;
    }



    /**
     * @var AudienceStateEventIdFlattenedDto
     */
    private $idFlattenedDto;

    /**
     * @return string
     */
    public function toString()
    {
        if (!$this->idFlattenedDto) {
            $this->idFlattenedDto = new AudienceStateEventIdFlattenedDto($this);
        }

        return $this->idFlattenedDto->toString();
    }

    /**
     * @param string $idStr
     *
     * @return AudienceStateEventId
     */
    public static function createFromString($idStr)
    {
        return (new AudienceStateEventIdFlattenedDto())
            ->fromString($idStr)
            ->toAudienceStateEventId();
    }


}

